ui/web.go

```diff
@@ -76,6 +77,11 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
 		reloadCh <- struct{}{}
 	})
 
+	r.Get("/-/healthy", func(w http.ResponseWriter, _ *http.Request) {
+		w.WriteHeader(http.StatusOK)
```

Comment:
could /healthy check that we can:
* read data from memory
* read data from disk

Reply:
Healthiness and readiness is often hard to distinguish, but I in my opinion you nailed it: anything regarding it's local operability defines it's healthiness. Readiness describes whether requests can be successfully served, meaning depending services can be reached, or in the mesh case, a mesh network was able to be established.

Reply:
wow, my comment was for prometheus, not alertmanager, sorry for the confusion.

For alertmanager we could just make sure that nothing is horribly wrong with the nflog or silences ?

Reply:
both of those points are applicable for Alertmanager as well though :)

Reply:
> For alertmanager we could just make sure that nothing is horribly wrong with the nflog or silences ?
Looking at the code I don't see anything that would stop silences and nflog from working once they have been instantiated. We could check that the maintenance hasn't failed for the last X calls but it seems a bit overkill to me.

Reply:
> We could check that the maintenance hasn't failed for the last X calls but it seems a bit overkill to me.

I want real metrics to alert on something like that, that happens at runtime. If instantiation is successful and there is nothing standing in the way of setting up the HTTP server to serve requests I'd say it's healthy. Readiness is to express that operationally we assume requests will actually successfully served including, basically if all other dependencies are "ready".